### PR TITLE
Windy leaves and water splash effects

### PR DESCRIPTION
**Pull request by pixel-stuck**

### CODE REVIEW

**Comment by BowserSlug** [comment]
🚀🚀🚀🚀🚀🚀 This PR matches 6 functions (+0.17%) on `us`.

File: src/effects/water_splash.c [hunk 1]
```diff
@@ -9,14 +11,145 @@ Gfx* D_E00BEAF0[] = {
...
+    ASSERT(part != NULL);
+
+    part->unk_00 = arg0;
+    part->unk_20 = 0;
+    if (arg5 <= 0) {
+        part->unk_1C = 1000;
+    } else {
+        part->unk_1C = arg5;
+    }
+    part->unk_30 = 0xFF;
```
Comment (ethteck) [inline src/effects/water_splash.c:58]: decimal for these

File: src/effects/water_splash.c [hunk 2]
```diff
@@ -9,14 +11,145 @@ Gfx* D_E00BEAF0[] = {
...
+    part++;
+
+    for(i = 1; i < effect->numParts; i++, part++) {
+        part->unk_04.x = 0.0f;
+        part->unk_04.y = 0.0f;
+        part->unk_04.z = 0.0f;
+        part->unk_34 = 2.0f;
+        switch (arg0) {
+            case 0:
+                part->unk_10.x = shim_rand_int(0xA) - 2;
```
Comment (ethteck) [inline src/effects/water_splash.c:75]: decimal for these

File: src/effects/windy_leaves.c [hunk 3]
```diff
@@ -1,14 +1,219 @@
...
+}
+
+void windy_leaves_update(EffectInstance *effect) {
+    f32 temp_f0_2;
+    f32 temp_f4;
+    s32 temp_v0;
+    s32 temp_v0_2;
+    s32 temp_a1;
+    s32 i;
+    WindyLeavesFXData *part = effect->data.windyLeaves;
```
Comment (ethteck) [inline src/effects/windy_leaves.c:88]: could you put this var at the top? also left pointer

File: src/effects/windy_leaves.c [hunk 4]
```diff
@@ -1,14 +1,219 @@
...
+    part->unk_2C++;
+    if (part->unk_28 < 0) {
+        shim_remove_effect(effect);
+        return;
+    }
+    temp_a1 = part->unk_2C;
+    temp_v0 = part->unk_28;
+    temp = part->type;
+    if (temp_a1 < 10) {
+        part->alpha += (0xFF - part->alpha) * 0.3;
```
Comment (ethteck) [inline src/effects/windy_leaves.c:102]: 255 -

File: src/effects/windy_leaves.c [hunk 5]
```diff
@@ -1,14 +1,219 @@
...
+                part->unk_14 += -0.05f;
+                part->unk_10 += temp3;
+                if (temp == 0) {
+                    part->unk_10 *= 0.94;
+                    part->unk_14 *= shim_sin_deg(part->unk_18) * 0.05 + 0.95;
+                } else {
+                    part->unk_14 += -0.05f;
+                    part->unk_10 *= 0.92;
+                }
+                part->unk_1C += func_E0200044(0x32, temp_a1 + i * 20) - 0x19;
```
Comment (ethteck) [inline src/effects/windy_leaves.c:122]: decimal plz

File: src/effects/windy_leaves.c [hunk 6]
```diff
@@ -1,14 +1,219 @@
...
+                part->unk_04.z += part->unk_38;
+            }
+            break;
+        case 1:
+            for(i = 1; i < effect->numParts; i++, part++) {
+                part->unk_10 += (f32)(shim_sin_deg(2.0f * part->unk_18) * 0.2 * 2.0);
+                part->unk_14 += -0.05f;
+                part->unk_10 *= 0.8464;
+                part->unk_14 += -0.05f;
+                part->unk_1C += func_E0200044(0x32, temp_a1 + i * 20) - 0x19;
```
Comment (ethteck) [inline src/effects/windy_leaves.c:166]: here2

File: src/effects/windy_leaves.c [hunk 7]
```diff
@@ -1,14 +1,219 @@
...
+                        part->unk_30 -= 0.8;
+                    } else {
+                        part->unk_30 += 0.8;
+                    }
+                    if (part->unk_04.z > 0.0f) {
+                        part->unk_38 -= 0.8;
+                    } else {
+                        part->unk_38 += 0.8;
+                    }
+                    part->unk_30 *= 0.9603999999999999;
```
Comment (ethteck) [inline src/effects/windy_leaves.c:182]: does 0.9604 work? if not, that snuckz

File: src/effects/windy_leaves.c [hunk 8]
```diff
@@ -1,14 +1,219 @@
...
+                    part->unk_38 *= 0.9603999999999999;
+                }
+                part->unk_04.x += part->unk_30;
+                part->unk_04.y += part->unk_34;
+                part->unk_04.z += part->unk_38;
+            }
+            break;
+        case 2:
+            for(i = 1; i < effect->numParts; i++, part++) {
+                part->unk_1C += func_E0200044(0x32,  temp_a1 + i * 20) - 0x19;
```
Comment (ethteck) [inline src/effects/windy_leaves.c:206]: dec

File: src/effects/windy_leaves.c [hunk 9]
```diff
@@ -1,14 +1,219 @@
...
+                part->unk_18 += shim_sin_deg(part->unk_1C) * 10.0f;
+                part->unk_20 += shim_cos_deg(part->unk_1C * 0.5) * 10.0f;
+                part->unk_04.x += part->unk_10;
+                part->unk_04.y += part->unk_14;
+            }
+            break;
+        default:
```
Comment (ethteck) [inline src/effects/windy_leaves.c:213]: no default necessary

**Review by ethteck — CHANGES_REQUESTED** [review]

**Review by ethteck — APPROVED** [review]

File: src/effects/water_splash.c [hunk 10]
```diff
@@ -9,14 +11,145 @@ Gfx* D_E00BEAF0[] = {
...
+        shim_remove_effect(effect);
+        return;
+    }
+
+    if (temp_a1_3 < 4) {
+        part->unk_30 = temp_a1_3 * 64;
+    }
+
+    part++;
+    for(i = 1; i < effect->numParts; i++, part++) {
```
Comment (nanaian) [inline src/effects/water_splash.c:134]:
```suggestion
    for (i = 1; i < effect->numParts; i++, part++) {
```

File: src/effects/water_splash.c [hunk 11]
```diff
@@ -33,4 +166,56 @@ void water_splash_render(EffectInstance* effect) {
...
+    shim_guTranslateF(sp10, sp94, temp_fp, data->unk_04.z);
+    shim_guScaleF(sp50, data->unk_34, data->unk_34, data->unk_34);
+    shim_guMtxCatF(sp50, sp10, sp10);
+    shim_guMtxF2L(sp10, &gDisplayContext->matrixStack[gMatrixListPos]);
+    gSPMatrix(gMasterGfxPos++, &gDisplayContext->matrixStack[gMatrixListPos++], G_MTX_PUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+    gSPMatrix(gMasterGfxPos++, currentCamera->unkMatrix, G_MTX_NOPUSH | G_MTX_MUL | G_MTX_MODELVIEW);
+    gSPDisplayList(gMasterGfxPos++, D_09000100_3BCB90);
+    
+    data++;
+    for(i = 1; i < ((EffectInstance*)effect)->numParts; i++, data++) {
```
Comment (nanaian) [inline src/effects/water_splash.c:190]: "

File: src/effects/windy_leaves.c [hunk 12]
```diff
@@ -1,14 +1,217 @@
...
+        s32 temp_s1 = func_E0200000(20);
+        s32 temp_s2 = func_E0200000(360);
+        s32 temp_v0_2 = func_E0200000(100);
+        part->unk_04.x = temp_s0_2 + 75;
+        part->unk_04.y = temp_s1 + 70;
+        part->unk_04.z = 0;
+        part->unk_18 = temp_s2;
+        part->unk_20 = 360 - temp_s2;
+
+        switch(type) {
```
Comment (nanaian) [inline src/effects/windy_leaves.c:50]: "

**Review by nanaian — CHANGES_REQUESTED** [review]
Good stuff, few syntax nits where we're missing a space after keywords, there are more than the ones I noted

**Pull request merged**